github-action/main.go

```diff
@@ -1,25 +1,108 @@
+	organizationName    = kingpin.Flag("org-name", "Qovery organization name").String()
+	projectId           = kingpin.Flag("project-id", "Qovery project ID").String()
+	projectName         = kingpin.Flag("project-name", "Qovery project name").String()
+	environmentId       = kingpin.Flag("env-id", "Qovery environment ID").String()
+	environmentName     = kingpin.Flag("env-name", "Qovery environment name").String()
+	applicationIds      = kingpin.Flag("app-ids", "Qovery application ID(s)").String()
+	applicationNames    = kingpin.Flag("app-names", "Qovery application name(s)").String()
 	applicationCommitId = kingpin.Flag("app-commit-id", "Application commit ID").String()
 	databaseId          = kingpin.Flag("db-id", "Qovery database ID").String()
+	databaseName        = kingpin.Flag("db-name", "Qovery database name").String()
 	apiToken            = kingpin.Flag("api-token", "Qovery API token").Required().String()
 )
 
+func getOrganizationId(qoveryAPIClient pkg.QoveryAPIClient, id *string, name *string) (string, error) {
+	if id != nil && *id != "" {
+		return *id, nil
+	}
+
+	if name != nil && *id != "" {
+		return qovery.GetOrganizationIdByName(qoveryAPIClient, *name)
+	}
+
+	return "", errors.New("'org-id' or 'org-name' property must be defined")
+}
+
+func getProjectId(qoveryAPIClient pkg.QoveryAPIClient, orgId string, id *string, name *string) (string, error) {
+	if id != nil && *id != "" {
+		return *id, nil
+	}
+
+	if name != nil && *id != "" {
+		return qovery.GetProjectIdByName(qoveryAPIClient, orgId, *name)
+	}
+
+	return "", errors.New("'project-id' or 'project-name' property must be defined")
+}
+
+func getEnvironmentId(qoveryAPIClient pkg.QoveryAPIClient, projectId string, id *string, name *string) (string, error) {
+	if id != nil && *id != "" {
+		return *id, nil
+	}
+
+	if name != nil && *id != "" {
+		return qovery.GetEnvironmentIdByName(qoveryAPIClient, projectId, *name)
+	}
+
+	return "", errors.New("'env-id' or 'env-name' property must be defined")
+}
+
+func getApplicationIds(qoveryAPIClient pkg.QoveryAPIClient, envId string, id *string, name *string) (string, error) {
+	if id != nil && *id != "" {
+		return *id, nil
+	}
+
+	if name != nil && *id != "" {
+		var ids []string
+		for _, sName := range strings.Split(*name, ",") {
+			id, err := qovery.GetApplicationIdByName(qoveryAPIClient, envId, sName)
+			catchError(err)
+
+			ids = append(ids, id)
+		}
+
+		return strings.Join(ids, ","), nil
+	}
+
+	return "", errors.New("'app-ids' or 'app-names' property must be defined")
+}
+
+func getDatabaseId(qoveryAPIClient pkg.QoveryAPIClient, envId string, id *string, name *string) (string, error) {
+	if id != nil && *id != "" {
+		return *id, nil
+	}
+
+	if name != nil && *id != "" {
+		return qovery.GetDatabaseIdByName(qoveryAPIClient, envId, *name)
+	}
+
+	return "", errors.New("'db-id' or 'db-name' property must be defined")
+}
+
+func catchError(err error) {
```

Comment:
OPT: Maybe rename this one to `handleError`, since it doesn't catch it but rather print and exit.

Reply:
makes sense. Thx